src/utils/fullOutcome.ts

```diff
@@ -0,0 +1,49 @@
+import Decimal from 'decimal.js';
+
+export interface FullOutcomeParams {
+  budget: string;
+  min: string;
+  max: string;
+  price: string;
+  buy?: boolean;
+}
+
+/**
+ * Performe geometric mean on a range of positive values
+ * @param min Should be lte max
+ * @param max Should be gte min
+ */
+export const geoMean = (min: string, max: string) => {
```

Comment:
TL;DR:
Usually I try to keep the `utils` folder dependent free of domain specific folder.
I'm happy to move both `fullOutcome.ts` & `fillOutcome.test.ts` files into the `strategies` folder if you want though :)

Why I've got this habit:
At this point in the life of this app it has no real impact, but as the app grows you'll want to build a clean dependency tree between services (to reduce build time for example) and you'll want the utils service to be a leaf of the tree to improve build/test performance.

Reply:
Sounds like a interesting topic for a presentation one day. Folder and File Structure Best Practices.

I am just afraid that if we start a new convention now it will get a bit messy. We can talk about a refactor and best practices to follow in the next sync. We should establish some code style guidelines.